### PR TITLE
chore: clean up react-doctor batch1 warnings

### PR DESCRIPTION
**Pull request by thienchi2109**
## Summary
- remove low-risk unused exports covered by Batch 1 and add export-surface guard tests
- eliminate the remaining React Doctor diff warnings in transfers and tenants management
- keep the TDD cleanup plan in docs for follow-up batches

## Test Plan
- verify:no-explicit-any
- typecheck
- focused vitest suite
- react-doctor diff scan

---

<a href="https://app.devin.ai/review/thienchi2109/qltbyt-nam-phong/pull/230" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

---
## Summary by cubic
Removes low‑risk unused exports and tightens module surfaces to reduce React Doctor Batch 1 warnings without behavior changes. Also simplifies transfers tab URL sync, adds accessible loading, and extracts a reusable tenant card to slim tenants management.

- **Refactors**
  - Page transitions: keep `MainContentTransition` public; make `PageTransitionWrapper`, `ModalTransition`, `LoadingTransition` internal; add export-surface test.
  - Transfers: de‑export `getTransferTypeConfig`; replace `useSearchParams` with window-based sync + `popstate`; route tabs through `TransfersSearchParamsBoundary` with accessible fallback; add export/source guard tests and boundary tests.
  - KPI barrel: remove unused type re‑exports; keep `KpiStatusBar` and status config exports; enforce via `src/components/kpi/index.exports.assertions.ts`.
  - Tenants management: extract `TenantsManagementTenantCard`; move role/type constants to `tenants-management-shared`; use `next/image`; add structure test to cap size and avoid ``.
  - Maintenance: minor import‑order cleanup.
  - Verification: focused export-surface and structure tests; `verify:no-explicit-any` and typecheck remain green; plan recorded at `docs/plans/2026-04-08-react-doctor-batch1-unused-exports-plan.md`.

<sup>Written for commit c2793223cb57cb82712abf638f5a63f74d90e36c. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * Added a tenant-management card component for per-tenant UI and actions.

* **Refactor**
  * Tenant management refactored to use shared role utilities and the new card.
  * Transfers UI routed through a dedicated search-params boundary and exports trimmed to reduce public API surface.

* **Bug Fixes**
  * Improved loading fallback accessibility for transfers.

* **Tests**
  * Added export-surface, structure, and source-level assertions; simplified boundary tests.

* **Documentation**
  * Added a controlled plan for cleaning up unused exports.

### CODE REVIEW

File: src/components/tenants-management.tsx [hunk 1]
```diff
@@ -1,463 +1,283 @@
...
+import { Input } from "@/components/ui/input"
+import { Button } from "@/components/ui/button"
+import { Skeleton } from "@/components/ui/skeleton"
+import { AddTenantDialog } from "@/components/add-tenant-dialog"
+import { EditTenantDialog, type TenantRow } from "@/components/edit-tenant-dialog"
+import { TenantsManagementTenantCard } from "@/components/tenants-management-tenant-card"
+import { useToast } from "@/hooks/use-toast"
+import { useSearchDebounce } from "@/hooks/use-debounce"
+
+const ROLE_LABELS = {
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/components/tenants-management.tsx]: Valid. Root cause was that the parent grouping logic and the extracted card each carried their own copy of the tenant role contract. I moved the shared role metadata and hierarchy types into src/components/tenants-management-shared.ts and imported that module from both src/components/tenants-management.tsx and src/components/tenants-management-tenant-card.tsx, so grouping and rendering now compile against the same source of truth. I also tightened src/components/__tests__/tenants-management.structure.test.ts to guard that shared import path.
